Assets/MixedRealityToolkit.SDK/Experimental/Features/UX/BoundsControl/BoundsControl.cs

```diff
@@ -431,15 +449,36 @@ private void Awake()
             if (targetObject == null)
                 targetObject = gameObject;
 
+            // ensure we have a default configuration in case there's none set by the user
+            scaleHandlesConfiguration = EnsureScriptable(scaleHandlesConfiguration);
+            rotationHandlesConfiguration = EnsureScriptable(rotationHandlesConfiguration);
+            boxDisplayConfiguration = EnsureScriptable(boxDisplayConfiguration);
+            linksConfiguration = EnsureScriptable(linksConfiguration);
+            handleProximityEffectConfiguration = EnsureScriptable(handleProximityEffectConfiguration);
+
+            // instantiate runtime classes for visuals
+            scaleHandles = new BoundsControlScaleHandles(scaleHandlesConfiguration);
+            rotationHandles = new BoundsControlRotationHandles(rotationHandlesConfiguration);
+            boxDisplay = new BoundsControlBoxDisplay(boxDisplayConfiguration);
+            links = new BoundsControlLinks(linksConfiguration);
+            proximityEffect = new ProximityEffect(handleProximityEffectConfiguration);
+
             // subscribe to visual changes
-            scaleHandles.configurationChanged.AddListener(CreateRig);
-            scaleHandles.visibilityChanged.AddListener(ResetVisuals);
-            rotationHandles.configurationChanged.AddListener(CreateRig);
-            boxDisplay.configurationChanged.AddListener(CreateRig);
-            links.configurationChanged.AddListener(CreateRig);
-
-            // subscribe visuals to bounds changes
-            boundsUpdated.AddListener(() => rotationHandles.CalculateEdgeCenters(ref boundsCorners));
+            scaleHandlesConfiguration.configurationChanged.AddListener(CreateRig);
```

Comment:
For CreateRig(), is there a work item to optimize that? So that on every call of configurationchanged, we aren't recreating everything? I know there was a todo marker

Reply:
yeah it's part of cleaning up - I'm trying not to open up not too many holes at once - ticket for rig changes is here https://github.com/microsoft/MixedRealityToolkit-Unity/issues/6260

Reply:
Awesome stuff